### PR TITLE
Dev-4593 Vertical Line Text Adjustment

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Shifts the height of the lines and text descending.

**Technical details:**

> • add function to svg line to get text height
> 
> • add effect to call create y scale everytime total text height changes
> 
> • update add spacing to look for if additional spacing is too small

**JIRA Ticket:**
[DEV-4593](https://federal-spending-transparency.atlassian.net/browse/DEV-4593)

**[Mockup](https://bahdigital.invisionapp.com/share/7HIADD8Q53D#/screens/296034818)**

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
- [x] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)

Reviewer(s):

- [x] Code review complete

### CODE REVIEW

File: src/js/components/sharedComponents/SVGLine.jsx [hunk 1]
```diff
@@ -209,7 +209,7 @@ export default class SVGLine extends Component {
                 tabIndex="0"
                 className={classname}
                 x={this.state[`${data}TextX`]}
-                y={this.state[`${data}TextY`]}
+                y={this.state[`${data}TextY`] || this.props.textY}
```
Comment (MaxwellKendall) [inline src/js/components/sharedComponents/SVGLine.jsx:212]: Why aren't these in initial state?

Also makes it really hard to know whats going on when we're dynamically accessing the state properties like this. Couldn't we just hard code it so we do `this.state.dataTextY/X`?

Seems strange also that we're using props some of the time and state some of the time?

File: src/js/components/award/shared/activity/ContractGrantActivityChartVerticalLines.jsx [hunk 2]
```diff
@@ -116,13 +128,14 @@ const ContractGrantActivityChartVerticalLines = ({
...
                 onMouseLeaveLine={showHideTooltip}
                 onMouseMoveText={showHideTooltip}
-                onMouseLeaveText={showHideTooltip} />}
+                onMouseLeaveText={showHideTooltip}
+                verticalLineTextHeight={verticalLineTextHeight} />}
             {/* potential end line */}
             {xScale && <SVGLine
                 scale={xScale}
-                y1={-10}
+                y1={potentialEndLineHeight - 10}
```
Comment (MaxwellKendall) [inline src/js/components/award/shared/activity/ContractGrantActivityChartVerticalLines.jsx:136]: Why 10?

File: src/js/components/sharedComponents/SVGLine.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/components/sharedComponents/SVGLine.jsx:212]: Yeah, this allows for developers to use multiple words and multiple text elements.  We use the props for the initial render.

File: src/js/components/award/shared/activity/ContractGrantActivityChartVerticalLines.jsx (hunk 2 above)
Comment (jonhill13) [reply src/js/components/award/shared/activity/ContractGrantActivityChartVerticalLines.jsx:136]: we extend the y axis line by ten

File: src/js/components/sharedComponents/SVGLine.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/sharedComponents/SVGLine.jsx:212]: This is an anti-pattern. We should be using either props or state but not both.

**Pull request closed without merging**